src/lib.rs

```diff
@@ -186,50 +193,170 @@ decl_module! {
+        <ClassById<T>>::insert(class_id, new_class);
+
+        // Increment the next class id:
+        <NextClassId<T>>::put(class_id + 1);
+
+        Self::deposit_event(RawEvent::ClassCreated(class_id));
+        Ok(class_id)
+    }
 
-        // let class_id = <NextClassId<T>>::get();
+    pub fn add_class_property(class_id: ClassId, property: Property) -> Result<u16, &'static str> {
 
-        // // Create new category
-        // let new_class = Class {
-        //     // TODO finish
-        // };
+        ensure!(<ClassById<T>>::exists(class_id), ERROR_CLASS_NOT_FOUND);
 
-        // // Insert category in map
-        // <ClassById<T>>::insert(new_class.id, new_class);
+        let mut class = <ClassById<T>>::get(class_id);
+        let prop_idx = class.properties.len() as u16;
+        class.properties.push(property);
+        <ClassById<T>>::insert(class_id, class);
 
-        // // Update other things
-        // <NextClassId<T>>::put(class_id + 1);
+        Self::deposit_event(RawEvent::ClassPropertyAdded(class_id, prop_idx));
+        Ok(prop_idx)
+    }
 
-        // // Generate event
-        // Self::deposit_event(RawEvent::ClassCreated(account.clone(), class_id));
+    pub fn add_class_schema(class_id: ClassId, schema: ClassSchema) -> Result<u16, &'static str> {
+        
+        ensure!(<ClassById<T>>::exists(class_id), ERROR_CLASS_NOT_FOUND);
 
-        Err(ERROR_NOT_IMPLEMENTED)
-    }
+        ensure!(schema.properties.len() > 0, ERROR_NO_PROPS_IN_CLASS_SCHEMA);
+        
+        let mut class = <ClassById<T>>::get(class_id);
 
-    pub fn add_class_property(class_id: ClassId, property: Property) -> Result<u16, &'static str> {
-        // TODO impl
-        Err(ERROR_NOT_IMPLEMENTED)
-    }
+        if let Some(prev_schema) = class.schemas.last() {
+            // Check that prev schema has a different set of props:
+            ensure!(prev_schema.properties != schema.properties, ERROR_SAME_PROPS_IN_CLASS_SCHEMA);
+        }
 
-    pub fn add_class_schema(class_id: ClassId, schema: ClassSchema) -> dispatch::Result {
-        // TODO impl
-        Err(ERROR_NOT_IMPLEMENTED)
+        // TODO check that schema contains props that are in the list of class props.
+
+        let schema_idx = class.schemas.len() as u16;
+        class.schemas.push(schema);
+        <ClassById<T>>::insert(class_id, class);
+
+        Self::deposit_event(RawEvent::ClassSchemaAdded(class_id, schema_idx));
+        Ok(schema_idx)
     }
 
-    pub fn create_entity(name: Vec<u8>, class_id: ClassId, schema_indices: Vec<u16>, property_values: Vec<(u16, PropertyValue)>) -> Result<EntityId, &'static str>{
-        // TODO impl
-        Err(ERROR_NOT_IMPLEMENTED)
+    pub fn create_entity(
+        class_id: ClassId,
+        schema_indices: Vec<u16>,
+        property_values: Vec<(u16, PropertyValue)>,
+        name: Vec<u8>
+    ) -> Result<EntityId, &'static str>{
+        
+        ensure!(<ClassById<T>>::exists(class_id), ERROR_CLASS_NOT_FOUND);
```

Comment:
See comment under `edit_entity`. It applies here also, the integrity check to see that `values` does indeed hold all required values for all claimed to be supported schemas (from `schemas`) will be quite complex, and there is no real need for it. Let's favor simplicity.

Reply:
Not sure what do you mean here. Maybe I lost context or it's fixed/refactored already.